### ISSUE TITLE
Inheritance

### ISSUE THREAD

**Issue by erikvanzijst** [aromanovich/jsl#13]
I'm trying to generate a schema that encodes inheritance using `allOf` in a subtype to merge the properties from a base type. However, I'm having trouble figuring out how to do this in jsl.

Is it possible to have jsl generate something along the following lines?

```javascript
    {
      "$ref": "#/definitions/circle",
      "definitions": {
        "shape": {
          "type": "object",
          "properties": {
            "color": {
              "type": "string"
            }
          }
        },
        "circle": {
          "type": "object",
          "allOf": [
            {
              "$ref": "#/definitions/shape"
            },
            {
              "properties": {
                "radius": {
                  "type": "number"
                }
              }
            }
          ]
        }
      }
    }
```

Note that I want to the schema to explicitly encode the relationship between `shape` and `circle`, instead of ending up with 2 independent objects where `circle` contains its own, inline copy of the `color` element.

**Comment by aromanovich** [comment]
I'm on it. I will release a new version in a couple of days with the support for ``allOf``-inheritance.

**Comment by erikvanzijst** [comment]
Sweet!

**Comment by aromanovich** [comment]
I've released the 0.1.4 version on PyPI. I introduces the notion of [inheritance modes](http://jsl.readthedocs.org/en/latest/tutorial.html#document-inheritance). All you need to generate such a schema is to set the `inheritance_mode` Circle's option to `jsl.ALL_OF`:

```python
>>> class Shape(jsl.Document):
...     class Options(object):
...         definition_id = 'shape'
...         title = 'Shape'
...     color = jsl.StringField(required=True)
...
>>> class Circle(Shape):
...     class Options(object):
...         definition_id = 'circle'
...         title = 'Circle'
...         inheritance_mode = jsl.ALL_OF
...     radius = jsl.NumberField(required=True)
>>> print json.dumps(Circle.get_schema(ordered=True), indent=4)
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "definitions": {
        "shape": {
            "type": "object",
            "title": "Shape",
            "properties": {
                "color": {
                    "type": "string"
                }
            },
            "required": [
                "color"
            ],
            "additionalProperties": false
        }
    },
    "allOf": [
        {
            "$ref": "#/definitions/shape"
        },
        {
            "type": "object",
            "title": "Circle",
            "properties": {
                "radius": {
                    "type": "number"
                }
            },
            "required": [
                "radius"
            ],
            "additionalProperties": false
        }
    ]
}
```
